src/libstd/sys/unix/ext/net.rs

```diff
@@ -1349,6 +1928,92 @@ impl UnixDatagram {
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// #![feature(unix_socket_ancillary_data)]
+    /// use std::os::unix::net::{UnixDatagram, SocketAncillary};
+    /// use std::io::IoSliceMut;
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let sock = UnixDatagram::unbound()?;
+    ///     let mut buf1 = [1; 8];
+    ///     let mut buf2 = [2; 16];
+    ///     let mut buf3 = [3; 8];
+    ///     let mut bufs = &mut [
+    ///         IoSliceMut::new(&mut buf1),
+    ///         IoSliceMut::new(&mut buf2),
+    ///         IoSliceMut::new(&mut buf3),
+    ///     ][..];
+    ///     let mut fds = [0; 8];
+    ///     let mut ancillary = SocketAncillary::with_capacity_for_fds(fds.len());
+    ///     let (size, sender) = sock.recv_vectored_with_ancillary_from(bufs, &mut ancillary)?;
+    ///     let size_fd = ancillary.get_fds(&mut fds[..]);
+    ///     println!("received {} bytes and {} fds from {:?}", size, size_fd, sender);
+    ///     Ok(())
+    /// }
+    /// ```
+    #[unstable(feature = "unix_socket_ancillary_data", issue = "none")]
+    pub fn recv_vectored_with_ancillary_from(
+        &self,
+        bufs: &mut [IoSliceMut<'_>],
+        ancillary: &mut SocketAncillary,
+    ) -> io::Result<(usize, SocketAddr)> {
+        unsafe {
+            let (count, addr) = recv_vectored_with_ancillary_from(&self.0, bufs, ancillary)?;
+            let addr = addr?;
+
+            Ok((count, addr))
+        }
+    }
+
+    /// Receives data and ancillary data from socket.
+    ///
+    /// On success, returns the number of bytes read.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// #![feature(unix_socket_ancillary_data)]
+    /// use std::os::unix::net::{UnixDatagram, SocketAncillary};
+    /// use std::io::IoSliceMut;
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let sock = UnixDatagram::unbound()?;
+    ///     let mut buf1 = [1; 8];
+    ///     let mut buf2 = [2; 16];
+    ///     let mut buf3 = [3; 8];
+    ///     let mut bufs = &mut [
+    ///         IoSliceMut::new(&mut buf1),
+    ///         IoSliceMut::new(&mut buf2),
+    ///         IoSliceMut::new(&mut buf3),
+    ///     ][..];
+    ///     let mut fds = [0; 8];
+    ///     let mut ancillary = SocketAncillary::with_capacity_for_fds(fds.len());
+    ///     let size = sock.recv_vectored_with_ancillary(bufs, &mut ancillary)?;
+    ///     let size_fd = ancillary.get_fds(&mut fds[..]);
+    ///     println!("received {} bytes and {} fds", size, size_fd);
+    ///     Ok(())
+    /// }
+    /// ```
+    #[unstable(feature = "unix_socket_ancillary_data", issue = "none")]
+    pub fn recv_vectored_with_ancillary(
+        &self,
+        bufs: &mut [IoSliceMut<'_>],
+        ancillary: &mut SocketAncillary,
+    ) -> io::Result<usize> {
+        unsafe {
+            let (count, addr) = recv_vectored_with_ancillary_from(&self.0, bufs, ancillary)?;
+            let _addr = addr?;
```

Comment:
I add this function because @joshtriplett suggests in a previous message that `send_vectored_to` and `send_vectored_fds_to` should be available and share a common implementation.

Reply:
I meant just  the `_addr` line.

Reply:
I change the line to `addr?;` in c1e653f446807320b01af5120a36fbc124c754e4. I want to check if the `addr` is correct.